labeler-module/pkg/labeler/labeler.go

```diff
@@ -408,5 +503,14 @@ func (l *Labeler) handlePodEvent(obj any) error {
 		return fmt.Errorf("failed to get driver label for node %s: %w", pod.Spec.NodeName, err)
 	}
 
-	return l.updateNodeLabels(pod.Spec.NodeName, expectedDCGMVersion, expectedDriverLabel)
+	// Detect Kata runtime on the node
+	expectedKataLabel, err := l.getKataLabelForNode(pod.Spec.NodeName)
```

Comment:
this should be in the node informer handler right? Do we need this in the `handlePodEvent`?